toolkit/tools/internal/customizationmacros/customizationmacros_test.go

```diff
@@ -0,0 +1,288 @@
+			// Create a temporary directory for testing
+			tempDir := t.TempDir()
+
+			// Define the test data
+			macroFileName := "test_macros"
+
+			err := AddMacroFile(tempDir, tc.macros, macroFileName, tc.customComments)
+
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+
+				// Verify the existence and contents of the macro file
+				expectedFilePath := filepath.Join(tempDir, "/usr/lib/rpm/macros.d", macroFileName)
+				actualContents, err := file.ReadLines(expectedFilePath)
+				assert.NoError(t, err)
+
+				assert.Equal(t, tc.expectedContents, actualContents)
+
+				// Ensure all lines are either empty, one of the macros, or have a "#" prefix
+				for _, line := range actualContents {
+					trimedLine := strings.TrimSpace(line)
+					if trimedLine == "" {
+						continue
+					}
+					if trimedLine[0] == '#' {
+						continue
+					}
+					if line[0] == '%' {
+						continue
+					}
+					assert.Fail(t, "unexpected line in macro file: "+line)
+				}
+			}
+		})
+	}
+}
+
+func TestAddCustomizationMacros(t *testing.T) {
+	// Define the test cases
+	testCases := []struct {
+		name                string
+		disableDocs         bool
+		DisableRpmLocales   bool
+		OverrideRpmLocales  string
+		expectError         bool
+		expectedDocMacro    string
+		expectedLocaleMacro string
+	}{
+		{
+			name:              "DisableDocs",
+			disableDocs:       true,
+			DisableRpmLocales: false,
+			expectError:       false,
+			expectedDocMacro:  "%_excludedocs 1",
+		},
+		{
+			name:                "DisableRpmLocales",
+			disableDocs:         false,
+			DisableRpmLocales:   true,
+			expectError:         false,
+			expectedLocaleMacro: "%_install_langs NONE",
+		},
+		{
+			name:                "DisableDocsAndLocales",
+			disableDocs:         true,
+			DisableRpmLocales:   true,
+			expectError:         false,
+			expectedDocMacro:    "%_excludedocs 1",
+			expectedLocaleMacro: "%_install_langs NONE",
+		},
+		{
+			name:              "EnableDocsAndLocales",
+			disableDocs:       false,
+			DisableRpmLocales: false,
+			expectError:       false,
+		},
+		{
+			name:                "OverrideRpmLocales",
+			disableDocs:         false,
+			DisableRpmLocales:   false,
+			OverrideRpmLocales:  "en:de:fr",
+			expectError:         false,
+			expectedLocaleMacro: "%_install_langs en:de:fr",
+		},
+		{
+			name:               "DisableRpmLocalesAndOverrideRpmLocales",
+			disableDocs:        false,
+			DisableRpmLocales:  true,
+			OverrideRpmLocales: "en_US",
+			expectError:        true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			tempDir := t.TempDir()
+			err := AddCustomizationMacros(tempDir, tc.disableDocs, tc.DisableRpmLocales, tc.OverrideRpmLocales)
+
+			if tc.expectError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+
+				// If neither are enabled, ensure no file
+				if !tc.disableDocs && !tc.DisableRpmLocales && tc.OverrideRpmLocales == "" {
```

Comment:
In general, a test helper shouldn't reimplement the product code. While this isn't a full reimplementation, it does seem like a partial reimplementation. So, it might be worth considering how to make the test case data more generic.

For example, perhaps the test cast data can contain a list of files that are expected to exist under the `/usr/lib/rpm/macros.d` directory and a (non-exhaustive) list of lines expected to be in each file.

Reply:
I ran into the same with the other tests as well, ended up with the list of expected results. I can do the same here.